### PR TITLE
feat(can): Api hardware can id script

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Adding very basic CAN driver and a script to show which devices are on the can bus.

closes RET-166

# Changelog

- Created `drivers.can_bus` package
- added `arbitration_id` module. This is exactly the same as `include/can/core/ids.hpp`. 
- added `CanDriver` with support for building the driver, sending, and receiving messages.
- added `scripts.identify` which sends (broadcast) a `device_info_request` message and shows the `device_info_response` messages as they return.

# Review requests

Output using pipette FW and can analyzer on MAC:
```
~/code/opentrons/hardware> pipenv run python -m hardware.scripts.identify --interface pcan --bitrate 250000
Received response from Node: pipette(20)
	Arbitration id: id: 0x18018800, parts: function_code: 0x0, node_id: 0x10, message_id: 0x3003
```

# Risk assessment

None

### CODE REVIEW

File: hardware/hardware/drivers/can_bus/arbitration_id.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+"""Can bus arbitration id."""
```
Comment (amitlissack) [inline hardware/hardware/drivers/can_bus/arbitration_id.py]: All the ids in here should be moved out of here and into a file in `shared_data`. Either a JSON/YAML file or this file PY as is.

The header file in ot3-firmware should be generated from single `shared_data` source of truth.

Not essential for this PR.

Comment (sfoster1) [reply hardware/hardware/drivers/can_bus/arbitration_id.py]: I'd quibble that `shared-data` is the right place; that's previously really only been used for robot<->app and app<->pd and so on, not really robot python <-> robot hardware (e.g. gcodes). It doesn't really make a difference though.

File: hardware/tests/drivers/can_bus/test_arbitration_id.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    c = ArbitrationId(id=arbitration_id)
+    assert c.parts == expected
+
+
+@pytest.mark.parametrize(
+    argnames=["expected", "parts"],
+    argvalues=[
+        [0x0, ArbitrationIdParts(function_code=0x0, node_id=0x0, message_id=0)],
+        [
+            0x1FFFFFFF,
```
Comment (DerekMaggio) [inline hardware/tests/drivers/can_bus/test_arbitration_id.py]: This converts to the same thing as the previous test because the padding is stripped right?

**Review by DerekMaggio — APPROVED** [review]

File: hardware/hardware/drivers/can_bus/arbitration_id.py (hunk 1 above)
Comment (amitlissack) [reply hardware/hardware/drivers/can_bus/arbitration_id.py]: What would you suggest as an alternative?

We could have the FW get the monorepo generate header file from an enum defined in this package.

File: hardware/tests/drivers/can_bus/test_arbitration_id.py (hunk 2 above)
Comment (amitlissack) [reply hardware/tests/drivers/can_bus/test_arbitration_id.py]: Yes

**Review by Laura-Danielle — APPROVED** [review]
Looks good to me.

**Pull request merged**